README.md

```diff
@@ -171,132 +180,78 @@ Typically finetunes of the base models below are supported as well.
-How to
-llama_print_timings:        load time =   576.45 ms
-llama_print_timings:      sample time =   283.10 ms /   400 runs   (    0.71 ms per token,  1412.91 tokens per second)
-llama_print_timings: prompt eval time =   599.83 ms /    19 tokens (   31.57 ms per token,    31.68 tokens per second)
-llama_print_timings:        eval time = 24513.59 ms /   399 runs   (   61.44 ms per token,    16.28 tokens per second)
-llama_print_timings:       total time = 25431.49 ms
-```
+- [Lucy's Labyrinth](https://github.com/MorganRO8/Lucys_Labyrinth) - A simple maze game where agents controlled by an AI model will try to trick you.
 
 </details>
 
-<details>
-<summary>Demo of running both LLaMA-7B and whisper.cpp on a single M1 Pro MacBook</summary>
+## Supported backends
 
-And here is another demo of running both LLaMA-7B and [whisper.cpp](https://github.com/ggerganov/whisper.cpp) on a single M1 Pro MacBook:
+| Backend | Target devices |
+| --- | --- |
+| [Metal](./docs/build.md#metal-build) | Apple Silicon |
+| [BLAS](./docs/build.md#blas-build) | All |
+| [BLIS](./docs/backend/BLIS.md) | All |
+| [SYCL](./docs/backend/SYCL.md) | Intel and Nvidia GPU |
+| [MUSA](./docs/build.md#musa) | Moore Threads MTT GPU |
+| [CUDA](./docs/build.md#cuda) | Nvidia GPU |
+| [hipBLAS](./docs/build.md#hipblas) | AMD GPU |
+| [Vulkan](./docs/build.md#vulkan) | GPU |
+| [CANN](./docs/build.md#cann) | Ascend NPU |
 
-https://user-images.githubusercontent.com/1991296/224442907-7693d4be-acaa-4e01-8b4f-add84093ffff.mp4
+## Building and usage
 
-</details>
+The main product of this project is the `llama` library. It's C-style interface can be found in [include/llama.h](include/llama.h).
+The project also produces several examples and tools that can be used to get started. There are a few ways to obtain the binaries:
+
+- Clone this repository and build locally, see [how to build](./docs/build.md)
+- On MacOS or Linux, install `llama.cpp` via [brew, flox or nix](./docs/install.md)
+- Use a Docker image, see [documentation for Docker](./docs/docker.md)
+- Download pre-built binaries from [releases](https://github.com/ggerganov/llama.cpp/releases)
+
+For more information, refer to [Build llama.cpp locally](./docs/build.md)
+
+### Obtaining and quantizing models
+
+The [Hugging Face](https://huggingface.co) platform hosts a [large amount of LLMs](https://huggingface.co/models?library=gguf&sort=trending) compatible with `llama.cpp` - simply search for the [GGUF](https://github.com/ggerganov/ggml/blob/master/docs/gguf.md) file format.
```

Comment:
```suggestion
The [Hugging Face](https://huggingface.co) platform hosts a [large number of LLMs](https://huggingface.co/models?library=gguf&sort=trending) compatible with `llama.cpp` - simply search for the [GGUF](https://github.com/ggerganov/ggml/blob/master/docs/gguf.md) file format.
```

Reply:
Strictly speaking this is not 100% correct since you could store arbitrary data in GGUF files (and some people unfortunately upload broken GGUFs immediately after a new model release). As of right now though you can reasonably assume that any GGUF model will run with llama.cpp. But we should keep in mind to update this if GGUF models for other projects ever become popular.

Reply:
Hmm right, I can see from the link above that flux gguf is one of the trending model, which is not compat with llama.cpp. But I think it's acceptable for now, as this link is here to help user find commonly used GGUF models.

Indeed, on HF hub, we do have a measure to filter only llama.cpp-compatible models. Maybe we could add a specific query parameter for it in the future (for example, `&compatible=llama-cpp`). CC @julien-c too!

Reply:
I've updated the wording to no imply that all GGUF files will work with `llama.cpp`. Feel free to improve this in the future.